src/gamblers_ruin.py

```diff
@@ -26,7 +26,7 @@ def find_nth_state(transition_matrix: np.ndarray,
     """
     # @ operator works for matrix multiplication in numpy
     # no need to cast to np.matrix
-    return np.linalg.matrix_power(transition_matrix, n) @ initial_state
+    return initial_state @ np.linalg.matrix_power(transition_matrix, n)
```

Comment:
Why did we do this? I don't understand!!!

Reply:
Matrix multiplication is not commutative and the new one is the correct operation.